src/js/base/api.js

```diff
@@ -165,6 +169,9 @@ ripe.Ripe.prototype._requestURL = function(url, options, callback) {
     if (contentType) {
         request.setRequestHeader("Content-Type", contentType);
     }
+
+    this.trigger("pre_request", request, options);

```

Comment:
maybe it's better to name this event `build_request` and then have a pre_request on `loadstart`